### PR TITLE
Test: Add cilium monitor in GuestBook Examples

### PR DESCRIPTION
**Pull request by eloycoto**
Hack until monitor PR is not in place to get monitor logs for the
failing test.

Related to #3462

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
@ianvernon  This Build has three greens for the GuestBook, I think that the problem was a race condition on delete Policies and things were in a bad state. I'm happy to merge this PR and check if we hit more problems in the following days. 

The monitor part is a hack, but I was waiting for this batch mode. The monitor changes will be ready next Monday
https://github.com/cilium/microscope/commit/27a36279c8fd58db2c4e46562eb2c5da9cac256c

**Comment by ianvernon** [comment]
@eloycoto what makes you think it was a race condition for deleting policies? Would be interested in seeing how you came to this conclusion so we can have it documented somewhere in case an issue like this happens again.

**Comment by ianvernon** [comment]
Do we need to include the monitor part in this case since it looks like this was an issue with respect to how we delete policies? I.e., could we merge the fixes to the test vs. the monitor part separately? I say this because I have found that running the monitor remotely with a context in the background and then cancelling the context does not actually kill the monitor process within the container; to me, it is undesirable to launch a process which we do not properly terminate within the Cilium pod. I can provide an example for this once I get a running setup again.

**Comment by eloycoto** [comment]
@ianvernon 

The policy was deleted by `kubectl.Delete`, but it didn't wait until endpoints are updated, so the status should be something like this: 
```
PolicyVersion: 1
Kubectl.Delete()
Kubectl.CiliumPolicyAction("newPolicy") (Policy increments from 1 to 2)
PolicyVersion: 2
```
The policy was not applied, should be version 3 in all endpoints, so things can be in a bad state. 

With this approach: 
```
PolicyVersion: 1
Kubectl.CiliumPolicyAction(delete)
PolicyVersion: 2 (All endpoints are now updated)
Kubectl.CiliumPolicyAction("newPolicy") 
PolicyVersion: 3
```

With this method, we ensure that the policy is updated in both Cilium instances. It's only my guess.

Yep, monitor code can be deleted.

**Comment by ianvernon** [comment]
@eloycoto makes sense, thanks! I suggest we remove the monitor code for now and merge the policy deletion-related changes you made here. If failure happens again we can investigate further.

**Comment by eloycoto** [comment]
I made the change and build it again ;-)

**Comment by eloycoto** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -545,18 +545,15 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
 			By("Apply policy to web")
-			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
-			_, err := kubectl.CiliumPolicyAction(
+			_, err = kubectl.CiliumPolicyAction(
 				helpers.KubeSystemNamespace, kubectl.ManifestGet(webPolicy),
 				helpers.KubectlApply, 300)
 			Expect(err).Should(BeNil(), "Cannot apply web-policy")
 
-			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 3, kubectl)
```
Comment (aanm) [inline test/k8sT/Policies.go:557]: Why remove this? Aren't the endpoints ready at this time?

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -566,16 +563,12 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
 			By("Apply policy to Redis")
-			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
 			_, err = kubectl.CiliumPolicyAction(
 				helpers.KubeSystemNamespace, kubectl.ManifestGet(redisPolicy),
 				helpers.KubectlApply, 300)
 
 			Expect(err).Should(BeNil(), "Cannot apply redis policy")
 
-			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 1, kubectl)
```
Comment (aanm) [inline test/k8sT/Policies.go:576]: Same here

File: test/k8sT/Policies.go (hunk 1 above)
Comment (ianvernon) [reply test/k8sT/Policies.go:557]: `CiliumPolicyAction` already waits, so there's no point in waiting again:

https://github.com/cilium/cilium/blob/cdfb40c8aa4c26b83061555984d9a65788fdb2ae/test/helpers/kubectl.go#L690

**Review by aanm — APPROVED** [review]

**Pull request merged**